tools/docker_constraints.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+"""Generates constraints file to pin dependency versions in Docker images.
+
+Removes hashes from supplied requirements file and merges it with existing
+constraints file. Versions in requirements take precedence over constraints.
+
+Generates interstitial with hashes stripped from requirements.
+"""
+
+import sys
+
+import pip_install
+import merge_requirements
+
+def main(hashed_reqs_in, unhashed_reqs_out,
+         constraints_in, constraints_out):
+    with open(unhashed_reqs_out, 'w') as fd:
+        fd.write(pip_install.remove_requirements_hashes(hashed_reqs_in))
```

Comment:
nit: I'm a little worried about us breaking this without realizing it.

I believe none of our tests in Travis currently run this file and I don't think we're really maintaining a stable API for our various scripts in tools so I'm worried someone might refactor `pip_install.py`, not realizing that this file was using `remove_requirements_hashes` and break everything.

I wonder if we might be able to avoid this by making a simple script whose sole purpose was to strip hashes. Files who needed that functionality could then call into the main function or shell out to the entire script.

What do you think?